dex/networks/btc/script.go

```diff
@@ -516,15 +534,26 @@ func InputInfo(pkScript, redeemScript []byte, chainParams *chaincfg.Params) (*Sp
 		sigScriptSize = 0
 		witnessWeight = RedeemP2WPKHInputWitnessWeight
 	case scriptType.IsP2SH():
+
 		// If it's a P2SH, the size must be calculated based on other factors.
-		// Start with the signatures.
-		sigScriptSize = (DERSigLength + 1) * scriptAddrs.NRequired // 73 max for sig, 1 for push code
+
+		// If the redeem script is p2ms, there is a leading OP_0, but we do not
+		// know if it is actually multisig since M and N could both be 1. TODO:
+		// change this test to use the redeem script's type!
+		if scriptAddrs.NRequired > 1 /* not good enough, could be 1-of-1! */ {
+			sigScriptSize++ // P2SM's leading OP_0
+		}
```

Comment:
Ah, yes `ParseScriptType` already checked the redeem script this way.  I was thinking it would require another parse of the redeem script, but it turns out it's already done.